lxd/storage/drivers/driver_ceph_utils.go

```diff
@@ -1265,3 +1265,75 @@ func (d *ceph) resizeVolume(vol Volume, sizeBytes int64, allowShrink bool) error
 
 	return err
 }
+
+// findLastCommonSnapshot finds the last common snapshot from the list of snapshots.
+// The exact snapshot can be discovered based on the list of snapshots that require a refresh.
+func (d *ceph) findLastCommonSnapshot(snapshots []Volume, refreshSnapshots []string) (Volume, bool) {
+	// If we have to refresh snapshots on the target volume, make sure that they are still
+	// in the same order as on the source volume.
+	// Start with trying to find the last common snapshot.
+	if len(snapshots) > 0 {
```

Comment:
Challenge accepted :) We can only iterate over the list of target volume snaps as this is the only list available in case of migration (on the target side) and I wanted to use the same logic for local refreshes and migration.

I have another approach and will add a unit test just to be sure we cover all scenarios.

Reply:
very good.

Reply:
Fixed and added a test case. I am surprised how "easy" it now looks compared to the version before but the unit tests doesn't lie :)